### PR TITLE
fix header dropdown click event and added hover effect

### PR DESCRIPTION
**Pull request by gmrzone**
fix header dropdown click event and added hover effect

### CODE REVIEW

**Comment by codyseibert** [comment]
@gmrzone What was wrong with the click event in the first place?  I just worked on this last night and it seemed to have worked fine, but maybe I missed something?

**Comment by gmrzone** [comment]
@codyseibert click event was on Link tag which did not include padding and the padding area was not clickable. After the changes the dropdown is clickable on Link tag as well as padding around it. you can see the file changes.

**Comment by codyseibert** [comment]
@gmrzone ahh ok I understand what the fix was now, sorry about that!  thanks!

**Review by codyseibert — APPROVED** [review]

**Comment by Mr-spShuvo** [comment]
Your code format is pretty messed up.
Run command `npm run format` or `yarn format` or use Prettier plugin to automatically format your code.
Then do a commit. @gmrzone  Can you do that?

**Review by Mr-spShuvo — CHANGES_REQUESTED** [review]
Your code format is pretty messed up.
Run command npm run format or yarn format or use Prettier plugin to automatically format your code.
Then do a commit. @gmrzone Can you do that?

File: frontend/src/Css/HeaderBar.css [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#header {
```
Comment (Mr-spShuvo) [inline frontend/src/Css/HeaderBar.css:1]: Why do we need `css/HeaderBar.css`?
We have styles/components/HeaderBar.css

File: frontend/src/Components/Header.js [hunk 2]
```diff
@@ -40,25 +41,40 @@ function Header() {
 
       {showNavigation && (
         <div ref={navigationRef} className="card" id="user--navigation">
-          <div className="user-navigation--item">
+          <div
+            className="user-navigation--item"
+            onClick={(e) => {
+              e.stopPropagation();
```
Comment (Mr-spShuvo) [inline frontend/src/Components/Header.js:47]: `e.stopPropagation();` <- handle it in `closeDropdown` method

File: frontend/src/Components/Header.js [hunk 3]
```diff
@@ -40,25 +41,40 @@ function Header() {
...
-              Profile
-            </Link>
+            <Link to={`/profile/${user.username}`}>Profile</Link>
           </div>
 
-          <div className="user-navigation--item">
+          <div
+            className="user-navigation--item"
+            onClick={(e) => {
+              e.stopPropagation();
```
Comment (Mr-spShuvo) [inline frontend/src/Components/Header.js:59]: `e.stopPropagation();` <- handle it in `closeDropdown`

File: frontend/src/Components/Header.js [hunk 4]
```diff
@@ -40,25 +41,40 @@ function Header() {
...
-              Settings
-            </Link>
+            <Link to={'/settings'}>Settings</Link>
           </div>
 
-          <div className="user-navigation--item">
+          <div
+            className="user-navigation--item"
+            onClick={(e) => {
+              e.stopPropagation();
```
Comment (Mr-spShuvo) [inline frontend/src/Components/Header.js:71]: `e.stopPropagation();` <- handle it in `closeDropdown`

**Review by Mr-spShuvo — CHANGES_REQUESTED** [review]
Make these changes

**Review by Mr-spShuvo — APPROVED** [review]

File: frontend/src/Components/Header.js [hunk 5]
```diff
@@ -15,7 +15,6 @@ function Header() {
   };
 
   const closeDropdown = (e) => {
```
Comment (Mr-spShuvo) [inline frontend/src/Components/Header.js:17]: Did you revert back the changes?

**Comment by Mr-spShuvo** [comment]
@gmrzone Try to improve onClick method

```js
const closeDropdown = (e, path) => {
  e.stopPropagation();
  setShowNavigation(false);
  history.push(path); 
};
```

**Review by codyseibert — APPROVED** [review]

**Comment by Mr-spShuvo** [comment]
I think I will merge them, and enhance method.

**Pull request merged**